pkg/database/openshiftclusters.go

```diff
@@ -120,6 +124,43 @@ func (c *openShiftClusters) Get(ctx context.Context, key string) (*api.OpenShift
 	}
 }
 
+// QueueLength returns OpenShiftClusters un-queued document count.
+// If error occurs, -1 is returned with error message
+func (c *openShiftClusters) QueueLength(ctx context.Context) (int, error) {
+	partitions, err := c.collc.PartitionKeyRanges(ctx, "OpenShiftClusters")
+	if err != nil {
+		return -1, err
+	}
+
+	var countTotal int
+	for _, r := range partitions.PartitionKeyRanges {
```

Comment:
have you been able to get a db with multiple partitionkeyranges to test this out?  I'd like to find out how we can do that to understand Cosmos better (needed for the bbm work too) - if you know how, please share :)

Reply:
Not yet. Now with cosmosdb merged and this closer to result, will try to do some replicable

Reply:
I've tested this with multiple PKRs and it works great.  A good way to test is to install https://docs.microsoft.com/en-us/azure/cosmos-db/local-emulator